module/User/src/Service/UserService.php

```diff
@@ -136,6 +153,16 @@ public function logout(): void
         $this->authService->clearIdentity();
     }
 
+    /**
+     * Get the current user entity as it is stored in the database.
+     */
+    public function getIdentity(): ?UserModel
+    {
+        $identity = $this->authService->getIdentity();
+
+        return $this->find($identity?->getId());
+    }
```

Comment:
It feels weird that this is able to return `null` but it does not break anything.

Reply:
There are cases when this is `null`, for example during API calls